static/js/setup.js

```diff
@@ -8,7 +8,7 @@ import * as util from "./util";
 // Miscellaneous early setup.
 export let password_change_in_progress = false;
 export let password_changes = 0;
-const xhr_password_changes = new WeakMap();
+export const xhr_password_changes = new WeakMap();
```

Comment:
Do we want to just move these variables and their setter function in a prep commit?  I'm not sure there's a reason for them to live in the `setup` module when the code they control has been moved.  

(And in general `setup.js` is the kind of file that we ideally want to end up deleting, since it does a very small set of mostly unrelated things)

Reply:
It's possible doing that sort of preparatory migration might also help with some of the node testing issues, too.

Reply:
Okay, makes sense.

Reply:
Actually, I think we will have to keep `export const xhr_password_changes` and 
```
    $(document).ajaxSend((event, xhr) => {
        xhr_password_changes.set(xhr, password_changes);
    });
```
in setup.js since we cannot move the above code block in channel and if we move `xhr_password_changes` to channel, we have to import `channel` in `setup` which will create a cyclic dependency which we don't want to create if we can avoid right?

I am working on fixing test.

Reply:
I don't think we need this to be registered via `ajaxSend` rather than just inline in the `channel` code.  See the commit I pushed attempting a more complete migration of the code as a reference.